### DISCUSSION TITLE
Suddenly getting `MQTT Unauthorized` on all OpenHASP plates

### DISCUSSION THREAD

**Discussion by erwin** [HASwitchPlate/openHASP#369] [Q&A] [answered]
I've had OpenHASP plates working well on v0.6.3 for ~ 6 months now. 5 plates.

Suddenly, my plates are having MQTT problems. Was there a change in Home Assistant to the way MQTT works?

When I telnet into the plate, I get:

```
Escape character is '^]'.

Welcome

        open____ _____ _____ _____
          |  |  |  _  |   __|  _  |
          |     |     |__   |   __|
          |__|__|__|__|_____|__|
        Home Automation Switch Plate
        Open Hardware edition v0.6.3-dev

[14:37:16.400][63408/65792  3][43124/43712  2] TELN: Client login from 192.XXX.XXX.XXX
[14:37:20.385][63576/72228 11][43124/43712  2] MQTT: Disconnected from broker, reconnection...
[14:37:20.390][63576/70044  9][43124/43712  2] MQTT: plate_42cbf4
[14:37:20.457][63576/71652 11][43124/43712  2] MQTT: Unauthorized
[14:37:25.390][63576/72228 11][43124/43712  2] MQTT: Disconnected from broker, reconnection...
<repeated ~10 x>
[14:38:00.392][63576/70224  9][43124/43712  2] MQTT: plate_42cbf4
[14:38:00.453][63576/70144  9][43124/43712  2] MQTT: Unauthorized
[14:38:00.458][63576/70320  9][43124/43712  2] MQTT: Retry count exceeded, rebooting...
[14:38:00.464][63576/70608  9][43124/43712  2] TELN: Closing session from 192.XXX.XXX.XXX
#Connection closed by foreign host.
```

1. I don't know why I'm seeing `plate_42cbf4` here in the output. This plate's name on the MQTT page is: `plate_office`.

![image](https://user-images.githubusercontent.com/11534217/185338905-b42901e5-8bbf-4775-b88e-0d7e08441c4c.png)

2. I've verified the Username `mqtt` and the Password are both set in Home Assistant Under the `/config/users` page:

![image](https://user-images.githubusercontent.com/11534217/185339487-617fc007-bc38-47d3-a79f-fefea7ee789f.png)

3. Also inside HA, on the `hassio/addon/core_mosquitto/config`, I also have setup:

```
logins:
  - username: mqtt
    password: << SAME PASSWORD >>
require_certificate: false
certfile: fullchain.pem
keyfile: privkey.pem
customize:
  active: false
  folder: mosquitto
```

I have tried many permutations of unplugging, resetting, rebooting both home assistant and the OpenHASP flashed lanbon plates, but no luck.

I even did a `Factory Reset` inside of the OpenHasp on device Menu, but still i get:

```
[14:51:55.298][68556/69156  0][41108/41728  2] MQTT: plate_42cbf4
[14:51:55.364][67200/69012  2][41108/41728  2] MQTT: Unauthorized
```

When I set this up months ago, I thought the plates were sending the actual device name I configured (in this case `plate_office`) rather than a randomly generated hex name... Not sure if that part matters.

The important thing - what do I need to do to:

1) verify that MQTT is otherwise working

2) get my OpenHASP plates to connect to MQTT again?

**Comment by fvanroie** [comment, marked as answer]
I'm running an up-to-date Home Assistant and I haven't seen this issue. If all plates started disconnecting at the same time, then the cause is most likely a broker configuration or update. Do you have other MQTT clients, like Tasmota, connecting to the broker?

I don't have any `logins` my Mosquitto broker options. I have configured my plates with a separate HA user on `Settings` > `People`.
Can you try using you HA username and password on the plate? Or try using [MQTT Explorer](https://mqtt-explorer.com/) on your PC and see which username/password works.

**Comment by erwin** [comment]
> I don't have any logins my Mosquitto broker options. I have configured my plates with a separate HA user on Settings > People.

Thank you @fvanroie that was it

When I tried MQTT Explorer, it failed to connect... So it was entirely an MQTT configuration issue.

Once I removed that and restarted, I was able to everything back to normal.
